junifer/datagrabber/tests/test_hcp.py

```diff
@@ -0,0 +1,142 @@
+        ("REST2", "RL", "rfMRI_REST2_RL_hp2000_clean.nii.gz"),
+        ("SOCIAL", "LR", "tfMRI_SOCIAL_LR_hp2000_clean.nii.gz"),
+        ("SOCIAL", "RL", "tfMRI_SOCIAL_RL_hp2000_clean.nii.gz"),
+        ("WM", "LR", "tfMRI_WM_LR_hp2000_clean.nii.gz"),
+        ("WM", "RL", "tfMRI_WM_RL_hp2000_clean.nii.gz"),
+        ("RELATIONAL", "LR", "tfMRI_RELATIONAL_LR_hp2000_clean.nii.gz"),
+        ("RELATIONAL", "RL", "tfMRI_RELATIONAL_RL_hp2000_clean.nii.gz"),
+        ("EMOTION", "LR", "tfMRI_EMOTION_LR_hp2000_clean.nii.gz"),
+        ("EMOTION", "RL", "tfMRI_EMOTION_RL_hp2000_clean.nii.gz"),
+        ("LANGUAGE", "LR", "tfMRI_LANGUAGE_LR_hp2000_clean.nii.gz"),
+        ("LANGUAGE", "RL", "tfMRI_LANGUAGE_RL_hp2000_clean.nii.gz"),
+        ("GAMBLING", "LR", "tfMRI_GAMBLING_LR_hp2000_clean.nii.gz"),
+        ("GAMBLING", "RL", "tfMRI_GAMBLING_RL_hp2000_clean.nii.gz"),
+        ("MOTOR", "LR", "tfMRI_MOTOR_LR_hp2000_clean.nii.gz"),
+        ("MOTOR", "RL", "tfMRI_MOTOR_RL_hp2000_clean.nii.gz"),
+    ],
+)
+def test_dataladhcp1200_datagrabber(
+    tasks: Optional[str],
+    phase_encodings: Optional[str],
+    expected_path_name: str,
+) -> None:
+    """Test datalad HCP1200 datagrabber.
+
+    Parameters
+    ----------
+    tasks : str
+        The parametrized tasks.
+    phase_encodings : str
+        The parametrized phase encodings.
+    expected_path_name : str
+        The parametrized expected path name.
+
+    """
+    configure_logging(level="DEBUG")
+    dg = DataladHCP1200(
+        tasks=tasks,
+        phase_encodings=phase_encodings,
+    )
+    # Set URI to Gin
+    dg.uri = URI
+    # Set correct root directory
+    dg._rootdir = "."
+    with dg:
+        # Get all elements
+        all_elements = dg.get_elements()
+        # Get test element
+        test_element = all_elements[0]
+        # Get test element data
+        out = dg[test_element]
+        # Asserts data type
+        assert "BOLD" in out
+        # Assert data file name
+        assert out["BOLD"]["path"].name == expected_path_name
+        # Assert data file path exists
+        assert out["BOLD"]["path"].exists()
+        # Assert data file path is a file
+        assert out["BOLD"]["path"].is_file()
+        # Assert metadata
+        assert "meta" in out
+        meta = out["meta"]
+        assert "element" in meta
+        assert "subject" in meta["element"]
+        assert test_element[0] == meta["element"]["subject"]
+
+
+@pytest.mark.parametrize(
+    "tasks, phase_encodings",
+    [
+        ("REST1", "LR"),
+        ("REST1", "RL"),
+        ("REST2", "LR"),
+        ("REST2", "RL"),
+        ("SOCIAL", "LR"),
+        ("SOCIAL", "RL"),
+        ("WM", "LR"),
+        ("WM", "RL"),
+        ("RELATIONAL", "LR"),
+        ("RELATIONAL", "RL"),
+        ("EMOTION", "LR"),
+        ("EMOTION", "RL"),
+        ("LANGUAGE", "LR"),
+        ("LANGUAGE", "RL"),
+        ("GAMBLING", "LR"),
+        ("GAMBLING", "RL"),
```

Comment:
Can you try a couple of tasks too? Also one task, two phase encodings?

```
(["REST1", "REST2"], ["LR", "RL"]),
("REST1", ["LR", "RL"]),
(["REST1", "REST2"], None),
```

Reply:
The cases you commented should be good for the test right? I mean, we just need a few cases.

Reply:
exactly

Reply:
Done.